client/src/pages/ThesisPage/ThesisPage.tsx

```diff
@@ -1,25 +1,38 @@
-import React from 'react'
 import { usePageTitle } from '../../hooks/theme'
 import ThesisConfigSection from './components/ThesisConfigSection/ThesisConfigSection'
 import ThesisInfoSection from './components/ThesisInfoSection/ThesisInfoSection'
 import ThesisProposalSection from './components/ThesisProposalSection/ThesisProposalSection'
 import ThesisWritingSection from './components/ThesisWritingSection/ThesisWritingSection'
 import ThesisAssessmentSection from './components/ThesisAssessmentSection/ThesisAssessmentSection'
 import ThesisFinalGradeSection from './components/ThesisFinalGradeSection/ThesisFinalGradeSection'
+import { useParams } from 'react-router-dom'
+import ContentContainer from '../../app/layout/ContentContainer/ContentContainer'
+import { Space } from '@mantine/core'
+import ThesisHeader from './components/ThesisHeader/ThesisHeader'
+import ThesisProvider from '../../contexts/ThesisProvider/ThesisProvider'
 
 const ThesisPage = () => {
-  // TODO: implement component
-  usePageTitle('Topic')
+  const { thesisId } = useParams<{ thesisId: string }>()
+
+  usePageTitle('Thesis')
 
   return (
-    <>
-      <ThesisConfigSection />
-      <ThesisInfoSection />
-      <ThesisProposalSection />
-      <ThesisWritingSection />
-      <ThesisAssessmentSection />
-      <ThesisFinalGradeSection />
-    </>
+    <ThesisProvider thesisId={thesisId} requireLoadedThesis>
+      <ContentContainer>
```

Comment:
(Optional)

```suggestion
      <ContentContainer style={{ display: 'flex', flexDirection: 'column', gap: '16px' }}>
```

Reply:
What would be the benefit of that (just curious)

Reply:
You would get rid of all the spacers and can achieve the same behavior with one line of code. Still in the end it's just personal taste I guess. :)

Reply:
It did not really work as it changed also gaps inside the accordions. I would leave it like this for now and come back to that at a later point